client/helper.go

```diff
@@ -219,3 +224,76 @@ func AskForConfirmation(prompt string, tries int) bool {
 	}
 	return false
 }
+
+func ReadInteger(prompt string, tries int, min int, max int) (bool, int) {
+	reader := bufio.NewReader(os.Stdin)
+	if tries <= 0 {
+		tries = 2
+	}
+
+	for ; tries > 0; tries-- {
+		fmt.Printf("%s [%d-%d]: ", prompt, min, max)
+
+		res, err := reader.ReadString('\n')
+		if err != nil {
+			log.Fatalf("error: %v", err)
+			continue
+		}
+
+		res = strings.TrimSpace(res)
+		if len(res) == 0 {
+			continue
+		}
+
+		val, err := strconv.Atoi(res)
+		if err != nil || val < min || val > max {
+			fmt.Printf("Invalid value. Must be between %d and %d.\n", min, max)
+			continue
+		}
+		return true, val
+	}
+
+	return false, -1
```

Comment:
This has a bit C style vibe, i think I'd also make this return an error just like below, e.g.

```go
func ReadInteger(prompt string, tries int, min int, max int) (bool, int) {
...


		return val, nil
	}

	return 0, fmt.Errorf("maximum tries exceeded")
...
```

Then on usage site it would be something along this:
```go
	idx, err := helper.ReadInteger("Select a user to reset the password for", 3, 1, len(users))
	if err != nil {
		fmt.Println("Aborted:", err)
		ExitWithError = true
		return
	}
```

Reply:
Good point, that's indeed better. It's an unfinished thought - the original signature was returning just `int`, then I realized I need to handle scenario when the tries are exceeded. I will change it to error.